core/src/main/scala/org/apache/spark/sql/SnappySession.scala

```diff
@@ -219,6 +219,10 @@ class SnappySession(_sc: SparkContext) extends SparkSession(_sc) {
   @transient
   private[sql] var currentKey: SnappySession.CachedKey = _
 
+  @transient
+  private[sql] def planCaching: Boolean = sessionState.conf.getConfString(
+    Property.PlanCaching.name, "true").toBoolean
+
```

Comment:
Why a def and not var? Overhead of conf read multiple times in every parse will further reduce parser performance. Already its performance is down by over 50% (in ColumnTableTest.compare parser performance) compared to 0.9 release which I was planning to look into soon. Yes, this means that "planCachingAll" flag will not effect existing sessions but too high an overhead just for an extreme config option that will be used very rarely, if at all. Please change to val.

Reply:
Yes I was not liking it. I thought if I put it as val then the expression won't get evaluated again and again. Will check.